src/Lecture2.hs

```diff
@@ -74,7 +84,10 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: (Ord a) => Int -> [a] -> (Maybe a, [a])
+removeAt index list
+           | index < 0 || index >= length list = (Nothing, list)
+           | otherwise = (Just $ last $ take (index + 1) list, (take index list ++ drop (index + 1) list))
```

Comment:
That's a correct implementation 👍🏻 
Alternatively, you can use a standard function [splitAt](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:splitAt) instead of writing manual recursion 🔃